lib/png/generate_png.js

```diff
@@ -0,0 +1,23 @@
+async (imgString) => {
+  const exportMaxWidth = 4000;
+  const tempImg = new Image();
+  const loadImage = () => {
+    return new Promise((resolve, reject) => {
+      tempImg.onload = (event) => resolve(event.currentTarget);
+      tempImg.onerror = () => {
+        reject("error loading string as an image");
+      };
+      tempImg.src = imgString;
+    });
+  };
+  const img = await loadImage();
+  const canvas = document.createElement("canvas");
+  canvas.width = Math.min(img.width, exportMaxWidth);
+  canvas.height = (canvas.width * img.height) / img.width;
```

Comment:
do u wanna use the img width or the potentially clipped canvas width?

Reply:
I'm not sure what you mean? if the canvas width is maxWidth then the canvasHeight needs to be adjusted accordingly too. This code is the same as in the exporter

Reply:
because of the Math.min, canvas.width and img.width might be different. just confirming that this is intentional the way u have it

Reply:
the image will be drawn onto the canvas and the canvas.width width, not the img.width, so it won't be clipped